### PR TITLE
add base_url and base_file_url to TelegramBotClient constructor parameter

### PR DESCRIPTION
**Pull request by naderAbolfazli**

### CODE REVIEW

File: src/Telegram.Bot/TelegramBotClient.cs [hunk 1]
```diff
@@ -31,9 +31,9 @@ public class TelegramBotClient : ITelegramBotClient
 
         private static readonly Update[] EmptyUpdates = { };
 
-        private const string BaseUrl = "https://api.telegram.org/bot";
+        private string BaseUrl = "https://api.telegram.org/bot";
 
-        private const string BaseFileUrl = "https://api.telegram.org/file/bot";
+        private string BaseFileUrl = "https://api.telegram.org/file/bot";
```
Comment (Avend544) [inline src/Telegram.Bot/TelegramBotClient.cs]: Man you have to change fields and mark them as readonly. You initizilithation BaseFileUrl and BaseUrl  in a constructor and when declare them.

**Comment by MihaZupan** [comment]
Naming should follow the current style (no underscores in the middle of the name).
New parameters should also not come before the httpClient parameter, as it's less likely to ever be used and a breaking change for no reason.

**Comment by poulad** [comment]
Hey @naderAbolfazli 
What's the reason for parametrizing base URLs?

**Comment by naderAbolfazli** [comment]
> Hey @naderAbolfazli
> What's the reason for parametrizing base URLs?

hey
it's provided in some other languages telegram sdk such as python so those sdk could use for another api that likes telegram bot api. for example, in bale messenger new bot api released based on telegram bot api so bot developer could use their bot in both with only change of base_url

**Comment by poulad** [comment]
In that case, we might want to set the base URL on the HTTP client and not introduce additional parameters to the constructor.

**Comment by tuscen** [comment]
@poulad Wouldn't that require having two separate http clients since Bot API have a different url for files?

I don't mind having both base urls as properties with public setters.

**Comment by MihaZupan** [comment]
> Wouldn't that require having two separate http clients since Bot API have a different url for files?

It could work for Telegram's api, setting the base as `"https://api.telegram.org/"`, but I don't think anything is gained since you still have to prepend `"bot/"` or `"file/bot/"` and other services may not share the same base for file urls.

+1 on public setters, anyone wanting to use this will surely be informed of having to change these values

**Comment by naderAbolfazli** [comment]
public setters are ok?

File: src/Telegram.Bot/TelegramBotClient.cs [hunk 2]
```diff
@@ -31,9 +31,9 @@ public class TelegramBotClient : ITelegramBotClient
 
         private static readonly Update[] EmptyUpdates = { };
 
-        private const string BaseUrl = "https://api.telegram.org/bot";
+        private readonly string BaseUrl = "https://tapi.bale.ai/";
```
Comment (ndobryanskyy) [inline src/Telegram.Bot/TelegramBotClient.cs]: seems like a malicious link

**Comment by MihaZupan** [comment]
Closing as the issue has not been resolved for a while.

**Pull request closed without merging**